sdfg_utils.js

```diff
@@ -1,5 +1,137 @@
 // Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
 
+/**
+ * Return the string UUID for an SDFG graph element.
+ *
+ * UUIDs have the form of G\S\N\E, where:
+ * G = Graph list id
+ * S = State ID (-1 for (nested) SDFGs)
+ * N = Node ID (-1 for States, SDFGs, and Edges)
+ * E = Edge ID (-1 for States, SDFGs, and Nodes)
+ *
+ * @param {*} element   Element to generate the UUID for.
+ *
+ * @returns             String containing the UUID
+ */
+function get_uuid_graph_element(element) {
+    let undefined_val = -1;
+    if (element instanceof State) {
+        return (
+            element.sdfg.sdfg_list_id + '/' +
+            element.id + '/' +
+            undefined_val + '/' +
+            undefined_val
+        );
+    } else if (element instanceof NestedSDFG) {
+        let sdfg_id = element.data.node.attributes.sdfg.sdfg_list_id;
+        return (
+            sdfg_id + '/' +
+            undefined_val + '/' +
+            undefined_val + '/' +
+            undefined_val
+        );
+    } else if (element instanceof Node) {
+        return (
+            element.sdfg.sdfg_list_id + '/' +
+            element.parent_id + '/' +
+            element.id + '/' +
+            undefined_val
+        );
+    }
+    return (
+        undefined_val + '/' +
+        undefined_val + '/' +
+        undefined_val + '/' +
+        undefined_val
+    );
+}
+
+function recursively_find_graph(graph, graph_id, ns_node=undefined) {
+    if (graph.node(0).sdfg.sdfg_list_id === graph_id) {
```

Comment:
strange :)

Reply:
Is there a prettier way to do it? Can't seem to see a more direct way..